components/director/internal/domain/package/converter.go

```diff
@@ -0,0 +1,180 @@
+package mp_package
+
+import (
+	"database/sql"
+	"encoding/json"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/str"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/internal/repo"
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/pkg/errors"
+)
+
+//go:generate mockery -name=AuthConverter -output=automock -outpkg=automock -case=underscore
+type AuthConverter interface {
+	ToGraphQL(in *model.Auth) *graphql.Auth
+	InputFromGraphQL(in *graphql.AuthInput) *model.AuthInput
+}
+
+type converter struct {
+	auth AuthConverter
+}
+
+func NewConverter(auth AuthConverter) *converter {
+	return &converter{
+		auth: auth,
+	}
+}
+
+func (c *converter) ToEntity(in *model.Package) (*Entity, error) {
+	if in == nil {
+		return nil, nil
+	}
+
+	defaultInstanceAuth, err := marshallDefaultInstanceAuth(in.DefaultInstanceAuth)
+	if err != nil {
+		return &Entity{}, err
```

Comment:
Instead of creating default Entity instance just return nil (and apply to all similar occurrences).
```suggestion
		return nil, err
```